### DISCUSSION TITLE
how to reverse proxy realtime websocket with parcel

### DISCUSSION THREAD

**Discussion by crapthings** [appwrite/appwrite#5073] [Q&A]
console: [HPM] Client disconnected
console: [HPM] Rewriting path from "/api/v1/realtime?project=63db4c420b091e7a2ff6&channels%5B%5D=teams" to "/v1/realtime?project=63db4c420b091e7a2ff6&channels%5B%5D=teams"
console: [HPM] GET /api/v1/realtime?project=63db4c420b091e7a2ff6&channels%5B%5D=teams ~> https://xxx/
console: [HPM] Upgrading to WebSocket

console: [HPM] WebSocket error: Error: read ECONNRESET
    at TCP.onStreamRead (node:internal/stream_base_commons:217:20) {
  errno: -54,
  code: 'ECONNRESET',
  syscall: 'read'
}

**Comment by stnguyen90** [comment]
@crapthings, thanks for creating this! 🙏🏼 Are you saying you're using [parcel](https://parceljs.org/) as your reverse proxy? If not, what are you using?

**Reply by crapthings** [reply]
yes i use parcel, just trying to figure out how to proxy to ws endpoint that appwrite provide

**Reply by crapthings** [reply]
.parcelrc
```json
{
  "extends": "@parcel/config-default",
  "resolvers": ["@parcel/resolver-glob", "..."],
  "reporters":  ["...", "parcel-reporter-static-files-copy"]
}

```

.proxyrc.js
```js
const { createProxyMiddleware } = require('http-proxy-middleware');

module.exports = function (app) {
  app.use(
    createProxyMiddleware('/api', {
      target: 'https://jsonplaceholder.typicode.com',
      changeOrigin: true,
      // secure: false,
      // logLevel: 'debug',
      pathRewrite: {
        '^/api': '',
      },
    })
  )
}
```
